cli/pcluster/config/hit_converter.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+from pcluster.config import mappings
+from pcluster.config.s3_param_types import S3Section
+
+
+class HitConverter:
+    """Utility class which takes care of ensuring backward compatibility with the pre-HIT configuration model."""
+
+    def __init__(self, pcluster_config):
+        self.pcluster_config = pcluster_config
+
+    def convert(self):
+        """
+        Convert the pcluster_config instance from pre-HIT to HIT configuration model.
+
+        Currently, the conversion is performed only if the configured scheduler is Slurm.
+        """
+        cluster_section = self.pcluster_config.get_section("cluster")
+        scheduler = cluster_section.get_param_value("scheduler")
+        queue_settings = cluster_section.get_param_value("queue_settings")
+
+        if scheduler == "slurm" and not queue_settings:
```

Comment:
This was intentional. The idea behind it is that the pcluster_config should be agnostic respect to the conversion logic, so that converter is free to evolve without client classes having to know how it works internally.

Reply:
I see your point. I'm still of the idea that the converter is responsible to convert while whoever applies the converter is responsible to decide whether it is needed or not. In any case I'm ok leaving it as is